jcl/src/java.base/share/classes/java/lang/Throwable.java

```diff
@@ -319,12 +313,29 @@ public void printStackTrace (PrintStream err) {
  *				The writer to write the walkback on.
  */
 public void printStackTrace(PrintWriter err) {
+	printStackTraceHelper(err);
+}
+
+/**
+ * Outputs representation of the receiver's
+ * walkback on the Appendable specified by the argument.
+ *
+ * @param	appendable Appendable 
+ *		The Appendable object to the walkback will be written.
+ */
+private void printStackTraceHelper(Appendable appendable) {
 	StackTraceElement[] stack;
-	stack = printStackTrace(err, null, 0, false);
+	HashSet<Throwable> exceptionChainSet = null;
+	try {
+		exceptionChainSet = new HashSet<Throwable>();
```

Comment:
Is a `HashSet` the right choice here?  Should it be an `IdentityHashMap`?  What semantics do you think are correct here - the exact same exception or one that is equal?

Reply:
Good catch! I believe reference-equality is what we need here, so `IdentityHashMap` should be the way to go. Let me update the commit to use that.